### DISCUSSION TITLE
GHCR

### DISCUSSION THREAD

**Discussion by sirhaffy** [crazy-max/diun#1081] [Q&A]
I have this docker compose setup:

  diun:
    container_name: diun
    image: crazymax/diun:latest
    volumes:
      - "./data:/data"
      - "/var/run/docker.sock:/var/run/docker.sock"
    environment:
      - "TZ=Europe/Paris"
      - "LOG_LEVEL=info"
      - "DIUN_PROVIDERS_DOCKER=true"
      - "DIUN_PROVIDERS_DOCKER_WATCHBYDEFAULT=true"
      - "DIUN_REGISTRIES_GHCRIO_USERNAME=sirhaffy"
      - "DIUN_REGISTRIES_GHCRIO_PASSWORD=SECRET_PAT"
    restart: always
    
    
And I get this error:

WRN Cannot get remote manifest error="cannot get image digest from HEAD request: unable to retrieve auth token: invalid username/password: unauthorized" image=ghcr.io/sirhaffy/njord:latest provider=docker

But the PAT works when I login (docker login ghcr.io -u sirhaffy -p PAT).

Can´t find much info on this, should it work?

**Comment by crazy-max** [comment]
Where did you find this key `DIUN_REGISTRIES_` ?

**Reply by sirhaffy** [reply]
Embarresing enough Chat GPT...

**Reply by sirhaffy** [reply]
Is there no functionallity to pull packages from private GHCR ? Would be awsome.

**Reply by crazy-max** [reply]
It's in the official docs: https://crazymax.dev/diun/config/regopts/
See also https://crazymax.dev/diun/config/#environment-variables

**Reply by sirhaffy** [reply]
Thank you @crazy-max ! Btw love the name and logo.
